### DISCUSSION TITLE
Count Restart Pod Deployment

### DISCUSSION THREAD

**Discussion by haddan-rocha** [argoproj/argo-cd#12726] [General]
Good morning guys,
I need to monitor the number of POD restarts through the ARGOCD event. How do I get this event?

![image](https://user-images.githubusercontent.com/42756247/223105215-caae84ac-f8e3-42a4-801a-fe63c18a72c2.png)

**Comment by alexmt** [comment]
Argo CD API creates K8S event every time when user restart Deployment/StatefulSet and perform any other [resource action](https://argo-cd.readthedocs.io/en/stable/operator-manual/resource_actions/).

Here sample event :

```json
{
    "metadata": {
        "name": "guestbook.174a45d9e2386e22",
        "namespace": "argocd",
        "annotations": {
            "dest-namespace": "guestbook",
            "dest-server": "http://kubernetes.default.svc"
        }
    },
    "involvedObject": {
        "kind": "Application",
        "namespace": "argocd",
        "name": "guestbook",
        "apiVersion": "argoproj.io/v1alpha1"
    },
    "reason": "ResourceActionRan",
    "message": "admin ran action restart on resource apps/Deployment/guestbook",
    "source": {
        "component": "argocd-server"
    }
}
```

So you would need to monitor k8s events and send notifications when matching even occurred. Argo CD does not have that functionality built-in but it can be achieve e.g. using [Argo Events](https://github.com/argoproj/argo-events) using [resource event source](https://github.com/argoproj/argo-events/blob/master/examples/event-sources/resource.yaml)

**Reply by haddan-rocha** [reply]
Hi Alexander Matyushentsev,


Thanks for the feedback !!
Could you let me know which POD has the event mentioned above?
